client/client.go

```diff
@@ -12,32 +13,89 @@ import (
 	"github.com/statechannels/go-nitro/client/engine/chainservice"
 	"github.com/statechannels/go-nitro/client/engine/messageservice"
 	"github.com/statechannels/go-nitro/client/engine/store"
+	"github.com/statechannels/go-nitro/protocols"
 	directfund "github.com/statechannels/go-nitro/protocols/direct-fund"
 	"github.com/statechannels/go-nitro/types"
 )
 
+// ChannelFundedEvents is returned in a channel when a state channel has been successfly funded.
+type ChannelFundedEvent struct {
+	ChannelId   types.Destination
+	ObjectiveId protocols.ObjectiveId
+}
+
 // Client provides the interface for the consuming application
 type Client struct {
-	engine  engine.Engine // The core business logic of the client
-	Address *types.Address
+	engine                 engine.Engine // The core business logic of the client
+	Address                *types.Address
+	allChannelFundedEvents chan ChannelFundedEvent
+	listeners              map[protocols.ObjectiveId]chan<- ChannelFundedEvent
 }
 
 // New is the constructor for a Client. It accepts a messaging service, a chain service, and a store as injected dependencies.
 func New(messageService messageservice.MessageService, chainservice chainservice.ChainService, store store.Store, logDestination io.Writer) Client {
 	c := Client{}
 	c.Address = store.GetAddress()
 	c.engine = engine.New(messageService, chainservice, store, logDestination)
-
+	c.allChannelFundedEvents = make(chan ChannelFundedEvent, 100)
+	c.listeners = make(map[protocols.ObjectiveId]chan<- ChannelFundedEvent)
 	// Start the engine in a go routine
 	go c.engine.Run()
 
+	// Start the event handler in a go routine
+	// It will listen for events from the engine and dispatch events to client channels
+	go c.handleEngineEvents()
+
 	return c
 }
 
+func (c *Client) registerListener(objectiveId protocols.ObjectiveId, listener chan<- ChannelFundedEvent) {
+	c.listeners[objectiveId] = listener
+}
+func (c *Client) removeListener(objectiveId protocols.ObjectiveId) {
+	l, ok := c.listeners[objectiveId]
+	if !ok {
+		log.Fatalf("Could not find listener for objective id %s", objectiveId)
+	}
+	close(l)
+	delete(c.listeners, objectiveId)
+}
+
+// handleEngineEvents is responsible for monitoring the ToApi channel on the engine.
```

Comment:
```suggestion
// handleEngineEvents is responsible for monitoring the ToApi chan on the engine.
```

Internally, we try to use this jargon to avoid confusion